webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -0,0 +1,150 @@
+import Message, { MessageType } from 'shared/components/Message';
+import Spinner from 'shared/components/Spinner';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { routeToReactUrl } from 'shared/utils/routes';
+import { Project, Workspace } from 'types';
+import handleError from 'utils/error';
+
+import css from './WorkspaceQuickSearch.module.scss';
+
+interface Props {
+  children: React.ReactChild;
+}
+
+const WorkspaceQuickSearch: React.FC<Props> = ({ children }: Props) => {
+  const [ searchText, setSearchText ] = useState<string>('');
+  const [ workspaceMap, setWorkspaceMap ] = useState<Map<Workspace, Project[]>>(new Map());
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ isModalVisible, setIsModalVisible ] = useState(false);
+  const canceler = useRef(new AbortController());
+
+  const fetchData = useCallback(async () => {
+    try {
+      const workspaceResponse = await getWorkspaces(
+        { limit: 0, sortBy: 'SORT_BY_NAME' },
+        { signal: canceler.current.signal },
+      );
+      const filteredWorkspaces = workspaceResponse.workspaces.filter((w) => !w.immutable);
+      const projectAPIList = filteredWorkspaces
+        .map((workspace) =>
+          getWorkspaceProjects(
+            { id: workspace.id, sortBy: 'SORT_BY_NAME' },
+            { signal: canceler.current.signal },
+          ));
+      const projectResponse = (await Promise.all(projectAPIList))
+        .map((project) => project.projects);
+
+      // Promise.all preserves the order
+      const tempWorkspaceMap: Map<Workspace, Project[]> = new Map();
+      for (const workspace of filteredWorkspaces.reverse()) {
+        const projects = projectResponse.pop();
+        tempWorkspaceMap.set(workspace, projects ?? []);
+      }
+      setWorkspaceMap(tempWorkspaceMap);
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject: 'Unable to fetch data.',
+        silent: false,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [ ]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(e.target.value);
+  };
+
+  const onShowModal = () => {
+    setIsModalVisible(true);
+    fetchData();
+  };
+
+  const onHideModal = () => setIsModalVisible(false);
+
+  const onClickProject = useCallback((project: Project) => {
+    routeToReactUrl(paths.projectDetails(project.id));
+    onHideModal();
+  }, []);
+
+  const onClickWorkspace = useCallback((workspaceId: number) => {
+    routeToReactUrl(paths.workspaceDetails(workspaceId));
+    onHideModal();
+  }, []);
+
+  const treeData: DefaultOptionType[] = useMemo(() => {
+    const data: DefaultOptionType[] = Array.from(workspaceMap)
+      .map(([ workspace, projects ]) => {
+        const treeChildren: DefaultOptionType[] = projects
+          .filter((project) => project.name.includes(searchText))
```

Comment:
https://user-images.githubusercontent.com/28383080/187017218-8fcc27e7-d8b9-47e7-8471-95b235fe9d56.mov

i guess this is not ideal, but i think it's fine if we change the text of the input to say "Enter a project..."

Reply:
that wasn't intended. but fixed the bug. 
realized the bug after assign you as a reviewer 😅